libp2p/protocols/kademlia/kademlia.nim

```diff
@@ -37,12 +172,25 @@ proc new*(
           buf = await conn.readLp(4096)
           msg = Message.decode(buf).tryGet()
 
-        # TODO: handle msg.msgType
+        case msg.msgType
+        of MessageType.findNode:
+          let targetIdBytes = msg.key.get()
+          let targetId = PeerId.init(targetIdBytes).tryGet()
+          let closerPeers = kad.rtable.findClosest(targetId.toKey(), DefaultReplic)
+          let responsePb = encodeFindNodeReply(closerPeers, switch)
+          await conn.writeLp(responsePb.buffer)
+
+          # Peer is useful. adding to rtable
+          discard kad.rtable.insert(conn.peerId)
+        else:
+          raise newException(LPError, "unhandled kad-dht message type")
     except CancelledError as exc:
       raise exc
     except CatchableError:
-      error "could not handle request",
-        peerId = conn.PeerId, err = getCurrentExceptionMsg()
+      discard
+      # TODO: figure out why this fails:
+      # error "could not handle request",
+      #   peerId = conn.PeerId, err = getCurrentExceptionMsg()
```

Comment:
Why not leave the error log?

Reply:
I'm not familiar with the context behind this. @richard-ramos would you be happy to chime in?